call-home/src/bin/callhome/collector/report_models.rs

```diff
@@ -33,25 +34,26 @@ pub(crate) struct Volumes {
 impl Volumes {
     /// Receives a openapi::models::Volumes object and returns a new report_models::volume object by
     /// using the data provided.
+    pub(crate) fn new(volumes: &[Volume], event_data: EventData) -> Self {
+        let volume_sizes = volumes.iter().map(|vol| vol.spec.size);
+        let volume_sizes = SortedExactSizeIter::from_iter(volume_sizes, NonDescending);
 
-    pub(crate) fn new(volumes: Vec<Volume>, event_data: EventData) -> Self {
-        let volumes_size_vector = get_volumes_size_vector(volumes.as_slice());
         Self {
-            count: volumes_size_vector.len() as u64,
-            max_size_in_bytes: get_max_value(volumes_size_vector.clone()),
-            min_size_in_bytes: get_min_value(volumes_size_vector.clone()),
-            mean_size_in_bytes: get_mean_value(volumes_size_vector.clone()),
-            capacity_percentiles_in_bytes: Percentiles::new(volumes_size_vector),
+            count: volume_sizes.len() as u64,
+            max_size_in_bytes: get_max_value(&volume_sizes),
+            min_size_in_bytes: get_min_value(&volume_sizes),
+            mean_size_in_bytes: get_mean_value(volume_sizes.clone()),
```

Comment:
can't we pass the ref here? or perhaps even just the volumes?

Reply:
The sum method consumes the iterator. But that is something I can work around. I'll avoid the clone here.

Reply:
Can't see that code here now, but might be able to just move it down to the last init of the struct

Reply:
Yeah, I've gotten rid of this clone.